src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -170,32 +172,68 @@ public async Task<IActionResult> ExecuteAsync(StoredProcedureRequestContext cont
         }
 
         /// <inheritdoc />
-        public JsonDocument? ResolveInnerObject(JsonElement element, IObjectField fieldSchema, ref IMetadata metadata)
+        public JsonElement ResolveObject(JsonElement element, IObjectField fieldSchema, ref IMetadata metadata)
         {
             PaginationMetadata parentMetadata = (PaginationMetadata)metadata;
+            if (parentMetadata.Subqueries.TryGetValue(QueryBuilder.PAGINATION_FIELD_NAME, out PaginationMetadata? paginationObjectMetadata))
+            {
+                parentMetadata = paginationObjectMetadata;
+            }
+
             PaginationMetadata currentMetadata = parentMetadata.Subqueries[fieldSchema.Name.Value];
             metadata = currentMetadata;
 
             if (currentMetadata.IsPaginated)
             {
                 return SqlPaginationUtil.CreatePaginationConnectionFromJsonElement(element, currentMetadata);
             }
-            else
-            {
-                //TODO: Try to avoid additional deserialization/serialization here.
-                return ResolverMiddleware.RepresentsNullValue(element) ? null : JsonDocument.Parse(element.ToString());
-            }
+
+            return element;
         }
 
-        /// <inheritdoc />
-        public object? ResolveListType(JsonElement element, IObjectField fieldSchema, ref IMetadata metadata)
+        /// <summary>
+        /// Resolves the JsonElement, an array, into a list of jsonelements where each element represents
+        /// an entry in the original array.
+        /// </summary>
+        /// <param name="array">JsonElement representing a JSON array. The possible representations:
+        /// JsonValueKind.Array -> ["item1","itemN"]
+        /// JsonValueKind.String (array of objects) -> [ { "field1": "field1Value", "field2": "field2Value" }, { ... } ]
+        /// The passed in JSON element's value kind may be an array or string.</param>
+        /// <param name="fieldSchema">Definition of field being resolved. For lists: [/]items:[entity!]!]</param>
+        /// <param name="metadata">PaginationMetadata of the parent field of the currently processed field in HC middlewarecontext.</param>
+        /// <returns>List of JsonElements parsed from the provided JSON array.</returns>
+        public object ResolveList(JsonElement array, IObjectField fieldSchema, ref IMetadata metadata)
```

Comment:
because when it returns JsonElement, the HC12 engine doesn't know how to handle the JsonElement. Michael had originally used JsonElement here, but that caused requests to fail during runtime.

Reply:
mention this in a code comment

Reply:
done. Added as comment.